cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/AppWithMapReduceUsingRuntimeDatasets.java

```diff
@@ -84,26 +83,21 @@ public void beforeSubmit(MapReduceContext context) throws Exception {
       String outputName = runtimeArgs.get(OUTPUT_NAME);
       String outputPath = runtimeArgs.get(OUTPUT_PATH);
 
-      FileSet input;
-      FileSet output;
       if (inputName.equals(outputName)) {
```

Comment:
It seems simpler to just replace the if and else blocks using the code from the else case.
Its the same code, only uses 1 less hashmap, but makes the code messier, harder to read.